pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/columnminmaxvalue/ColumnMinMaxValueGenerator.java

```diff
@@ -56,6 +56,34 @@ public ColumnMinMaxValueGenerator(SegmentMetadataImpl segmentMetadata, SegmentDi
     _columnMinMaxValueGeneratorMode = columnMinMaxValueGeneratorMode;
   }
 
+  public boolean needAddColumnMinMaxValue() {
+    Schema schema = _segmentMetadata.getSchema();
+    Set<String> columnsToAddMinMaxValue = new HashSet<>(schema.getPhysicalColumnNames());
```

Comment:
Will the code duplication be fixed up in follow-up PR (when the flow is actually changed) where this logic of "whether update is needed or not" is removed from `addColumnMinMaxValue`() function and latter will simply add ?

Reply:
will remove some duplicate code here firstly.

> ... addColumnMinMaxValue() function and latter will simply add?

Probably not (if I understand this question correctly). Because that would require the needProcess() to check all conditions and collect what's to be added/removed/updated and pass that info to process() method later on. To begin with, needProcess() returns a boolean, and stops as soon as it finds a need for process, for simplicity.